Wikipedia/Code/WMFAppViewController.m

```diff
@@ -207,14 +220,74 @@ - (void)pauseApp {
     [self performHousekeepingIfNecessary];
 }
 
+- (void)handleSelectionOfShortcutItemIfNecessaryThen:(void (^)(BOOL shortcutWasHandled))completion {
+    UIApplicationShortcutItem* shortcutItemSelectedAtLaunch = self.shortcutItemSelectedAtLaunch;
+    if (shortcutItemSelectedAtLaunch) {
+        // Main UI may *not* have been loaded if the user...
+        //      - minimized the app when viewing a "Welcome" screen
+        //      - then selected a 3d touch app icon shortcut menu item
```

Comment:
they won't see the shortcut items since they need to be added at runtime.
However, they could minimize the app during on-boarding and *then* try to
use a shortcut item. side note: users can also circumvent on-boarding by
force-quitting and relaunching, but that's not really related to this.

On Tue, Jan 12, 2016 at 1:39 PM, Corey Floyd <notifications@github.com>
wrote:

> In Wikipedia/Code/WMFAppViewController.m
> <https://github.com/wikimedia/wikipedia-ios/pull/319#discussion_r49494519>
> :
>
> > @@ -207,14 +220,74 @@ - (void)pauseApp {
> >      [self performHousekeepingIfNecessary];
> >  }
> >
> > +- (void)handleSelectionOfShortcutItemIfNecessaryThen:(void (^)(BOOL shortcutWasHandled))completion {
> > +    UIApplicationShortcutItem* shortcutItemSelectedAtLaunch = self.shortcutItemSelectedAtLaunch;
> > +    if (shortcutItemSelectedAtLaunch) {
> > +        // Main UI may *not* have been loaded if the user...
> > +        //      - minimized the app when viewing a "Welcome" screen
> > +        //      - then selected a 3d touch app icon shortcut menu item
>
> If the user hasn't finished viewing the welcome screen, should there be
> any actions in the short cut menu?
> Users probably shouldn't be able to use the shortcut menu to "skip" the
> welcome screen.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wikimedia/wikipedia-ios/pull/319/files#r49494519>.
>



-- 
EN Wikipedia user page: https://en.wikipedia.org/wiki/User:Brian.gerstle
IRC: bgerstle